client/src/model/values/instantaneousValues/vectorValues/VectorValue.hx

```diff
@@ -0,0 +1,148 @@
+package model.values.instantaneousValues.vectorValues;
+import model.values.instantaneousValues.vectorValues.VectorValueI;
+import model.values.instantaneousValues.InstantaneousValueI;
+import model.values.instantaneousValues.scalarValues.ScalarValueI;
+
+/**
+ * ...
+ * @author AdvaitTrivedi
+ */
+class VectorValue implements VectorValueI 
+{
+	var vector : Array<InstantaneousValueI> = new Array<InstantaneousValueI>();
+
+	public function new() 
+	{
+		
+	}
+	
+	
+	/* INTERFACE model.values.VectorValueI */
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function fromArray(instanteousValues: Array<InstantaneousValueI>) : VectorValueI {
+		for (value in instanteousValues) {
+			this.push(value);
+		}
+		
+		return this;
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function push(instantaneousValue:InstantaneousValueI, ?index:Int = 0):Void 
+	{
+		this.vector.insert(index, instantaneousValue);
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function removeFrom(?index:Int = 0):InstantaneousValueI 
+	{
+		if (index == 0) {
+			return this.vector.shift();
+		}
+		
+		if (index > this.length()) {
+			throw "Invalid index";
+		}
+		
+		var valuePopped = this.vector[index];
+		this.vector.remove(valuePopped);
+		return valuePopped;
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function concat(vectorValue:VectorValueI):Void 
```

Comment:
I don't think we'll need this method.  A functional version of concat might be useful, though.

Reply:
See comments above on mutators.